server/src/core/search/search_executor/postprocess.rs

```diff
@@ -0,0 +1,270 @@
+use super::meilisearch;
+use super::preprocess;
+
+pub(super) fn merge_search_results(
+    args: super::SanitisedSearchQueryArgs,
+    search_tokens: &Vec<preprocess::SearchToken>,
+    res_merged: meilisearch::MSResults,
+    res_buildings: meilisearch::MSResults,
+    res_rooms: meilisearch::MSResults,
+) -> Vec<super::SearchResultsSection> {
+    // First look up which buildings did match even with a closed query.
+    // We can consider them more relevant.
+    let mut closed_matching_buildings = Vec::<String>::new();
+    for hit in res_buildings.hits {
+        closed_matching_buildings.push(hit.id);
+    }
```

Comment:
I am also wondering why. Maybe because it doesn't know that `.push()` shouldn't have side-effects.
But anyway, this code is unused to my knowledge. We could throw it away for now.

Reply:
I dont think that this is a good idea. adding a todo would probably be better.
This is the only place where `res_buildings` is used.
This does seem like a bug to me..

Reply:
added a TODO